### DISCUSSION TITLE
Flter module for admn view (or custom view)?

### DISCUSSION THREAD

**Discussion by cpaschen** [joomengine/Joomla-Component-Builder#864] [Q&A] [answered]
I'm trying to add a search/filter function to a custom component created in JCB, but I'm not exactly sure even where to start.

I have a table of product parts that have (among others) the columns "part_type" and "engine_type".
I also have a separate table for part_type and engine_type.

I need to have a module on the site that allows users select a PART TYPE and ENGINE TYPE (via select box) and when clicking a SEARCH button, a custom list view of all patching parts is displayed.
Here's a sample of what the module should look like:
![part_search_module](https://user-images.githubusercontent.com/1929923/152468917-33b45fa2-f1d0-48d4-8b2b-6b03edb72cf5.png)

However, I'm not sure if there is a way to build this within JCB, or if there is any similar example within any of the JCB demos.

I've also reviewed the J3 smart search, but I don't see any way to make this work by searching (i.e. filtering) on 2 different fields. If that is possible please let me know and I'll try to dig more into that.

ANY help would be appreciated.

BTW ... the table is huge, over 50,000 parts so it would help knowing what would be the most 'efficient' (i.e. fast) way to run the filtering/search process.

**Comment by cpaschen** [comment]
Also, that search/filter box/module must appear on the main page and then the results need to be displayed on a new page after clicking the Search button (we can't just create a page of parts with filtering on them).

And, I started to manually build a Joomla module with a form and try to do this all manually, but wondering if there is an easier (JCB) way to create this functionality.

**Comment by TLWebdesign** [comment]
Hi Chris, have you looked at these tutorials: https://www.youtube.com/channel/UCYDNg3vgWGL7CRtWFNcFqew They show you how to build a filter system on a front-end view.

**Comment by TLWebdesign** [comment]
I think it be wise to create a "search" front-end view so you have that seperate from a normal product page view.

**Comment by cpaschen** [comment]
Yes, I've used those videos to learn how to do searching/filtering in the past; however, this use case requires that the filter fields be displayed on a different page (home page; in a module area) than the list view.
So I can't use the method he uses there, where the filter fields are displayed via js on the same page and just update the page.

My best guess is that I would create a form in a module that, when completed and button clicked, will pass the filter parameters to the dynamic get/list view page and display it (I'm guessing via POST method). But I'm not exactly sure how to get the data transferred from the module form to the dynamic get/list view in order to filter the list.

Unfortunately, per client specs, the filter display must be designed like this (filter on home page).

I did go back and re-watch a few of those videos but I didn't see any example of having the filters on a different page. If I missed that please point me to the video with that explanation.

Otherwise, anyone have any ideas where to find how to use POST values as the inputs to custom view/dynamic gets?

**Comment by cpaschen** [comment]
BTW ... there was a post that looked similar here: https://github.com/vdm-io/Joomla-Component-Builder/discussions/824
But no response was given.

**Comment by cpaschen** [comment]
OR is there a way to use session variables to filter a site list view?

**Comment by cpaschen** [comment]
Wait, did another search here and found this: https://github.com/vdm-io/Joomla-Component-Builder/discussions/654
Which I think might provide some help.

I'll update here after I've tried a few other options.

**Reply by TLWebdesign** [reply]
You can use jinput in your dynamic get to get the value you posted to the page. I've done that the other day when i needed a button to be clicked and filter the page. 

<img width="1558" alt="Schermafbeelding 2022-02-07 om 08 08 01" src="https://user-images.githubusercontent.com/4402824/152741047-09b93128-45e4-4f49-aa0f-d30a8b18aa53.png">

Another way is using session variables and local storage. That way your selection can be persistent. Very nice way of doing it. You just do the filtering in the custom script section of your dynamic get.

**Comment by dormasta** [comment]
Hi,
I don't know if it helps, but I found some documentation about the smart search plugin which needs to be extended for your component. Btw. I couldn't test it yet because the "smart search" somehow doesn't work with my template at all for the moment.

https://docs.joomla.org/Creating_a_Smart_Search_plug-in
https://magazine.joomla.org/all-issues/issue-february-2012/Developing-a-Smart-Search-Plugin

**Comment by cpaschen** [comment, marked as answer]
Thanks everyone.
I was able to make this work rather easily.
AFTER I started to hand-code a custom form (in a module), and getting tired trying to do that by hand, I realized (by work on another project) that the Convert Forms by Tassos Marinos (https://www.tassos.gr/joomla-extensions/convert-forms) allows creating a pretty custom form in a module. The form can redirect to another page on submit and pass the selections.
This Convert Forms set-up for the interface, and the details provided by @TLWebdesign made it a very quick solution

Hope this provides some help to anyone else needing to do something like this in the future.
